libnavui-shield/src/main/java/com/mapbox/navigation/ui/shield/ShieldsCache.kt

```diff
@@ -187,7 +187,7 @@ internal class ShieldResultCache(
         val shieldUrl = toDownload.url
         return shieldByteArrayCache.getOrRequest(shieldUrl).mapValue { byteArray ->
             RouteShield.MapboxLegacyShield(
-                toDownload.url,
+                toDownload.originalUrl,
```

Comment:
Let's add a `MapboxLegacyShield#initialUrl` for the purpose of matching to maneuver object. This way the `MapboxLegacyShield#url` can remain a valid executable URL. And please comment why do we need the initial URL and how it can be used to match with the maneuver.